app/components/nextflow/samplesheet/row_component.rb

```diff
@@ -6,68 +6,107 @@ module Samplesheet
     class RowComponent < Component
       attr_reader :sample, :files, :properties
 
-      def initialize(sample, index, properties)
+      def initialize(sample, index, properties, required_properties)
         @sample = sample
         @index = index
-        @properties = properties
-        @files = filter_files
-        @file_index = 0 # Keeps track of which file input is currently being rendered
+        @properties = format_properties(properties)
+        @required_properties = required_properties
+        @files = filter_paired_files
       end
 
-      def filter_files # rubocop:disable Metrics/AbcSize
-        first = []
-        second = []
+      def format_properties(properties)
+        properties.map do |name, entry|
+          entry['is_fastq'] = name.match(/fastq_\d+/)
+          next unless check_for_file(entry)
 
-        pattern = @properties['pattern'].to_s
+          entry['pattern'] = entry['pattern'] || entry['anyOf'].select do |e|
+            e.key?('format') && e['format'] == 'file-path'
+          end.pluck('pattern').join('|')
+        end
+        properties
+      end
 
-        @sample.attachments.each do |attachment|
-          # Check that the file meets the requirements for the pipeline
-          next unless pattern.nil? || attachment.file.filename.to_s.match(/#{Regexp.new(pattern)}/)
+      def filter_paired_files # rubocop:disable Metrics/MethodLength
```

Comment:
I think this method needs a different name since we aren't just filtering the paired end files

Reply:
Good catch, thanks.  Updated in b2b4519